### PR TITLE
Update to algoliasearch 2.0

### LINKED ISSUE #18: Upgrade to algoliasearch 2.0
I've been contacted by @nunomaduro to try the new algoliasearch 2.0.0b2 on this package.

Here is the gist of the new API usage: https://gist.github.com/nunomaduro/c59af518ef9d4ead937ba8310544d11c

### PR DESCRIPTION
**Pull request by Natim**
Fixes #18 
r? @nunomaduro

### CODE REVIEW

File: kinto_algolia/indexer.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def join(self):
         self.tasks = []
 
     def set_extra_headers(self, headers):
-        self.client.set_extra_headers(**headers)
+        self.client._config.headers.update(headers)
```
Comment (Natim) [inline kinto_algolia/indexer.py]: That's the simplest way I found to not recreate the client each time.

Another solution would be to keep the application_id and api_key and recreate the client there.

**Comment by Natim** [comment]
@nunomaduro: What would you suggest to replace is_alive here? https://github.com/Kinto/kinto-algolia/pull/28/files#diff-b4035ea4376fe730a3020b7b36640a14R129

File: kinto_algolia/indexer.py [hunk 2]
```diff
@@ -74,19 +73,19 @@ def search(self, bucket_id, collection_id, **kwargs):
...
-                res = index.clear_index()
-                self.tasks.append((indexname, res["taskID"]))
-                res = self.client.delete_index(indexname)
-                self.tasks.append((indexname, res["taskID"]))
-        self.join()
+                index.clear_objects().wait()
+                index.delete().wait()
+
+    def isalive(self):
+        self.client._transporter.read(Verb.GET, "1/isalive", {}, None)
```
Comment (Natim) [inline kinto_algolia/indexer.py]: That's eventually how I did it, but I guess it would have been better to keep the `isalive()` method.

Comment (nunomaduro) [reply kinto_algolia/indexer.py]: @Natim Sounds good at the moment, as this moment is not in our plans for the v2.

File: tests/test_includeme.py [hunk 3]
```diff
@@ -37,7 +37,7 @@ def test_present_in_heartbeat(self):
 
     def test_returns_false_if_connection_fails(self):
         with mock.patch.object(self.app.app.registry.indexer, "client") as client:
-            client.is_alive.side_effect = AlgoliaException
+            client._transporter.read.side_effect = AlgoliaException
```
Comment (nunomaduro) [inline tests/test_includeme.py]: This `side_effect` property doesn't exist. Is this test passing?

File: kinto_algolia/indexer.py (hunk 1 above)
Comment (nunomaduro) [reply kinto_algolia/indexer.py]: Sounds fine.

File: tests/test_includeme.py (hunk 3 above)
Comment (Natim) [reply tests/test_includeme.py]: It is because client has been mocked on line 39 here.

File: kinto_algolia/indexer.py (hunk 2 above)
Comment (Natim) [reply kinto_algolia/indexer.py]: Why did you choose to remove it?

**Pull request merged**